src/main/java/seedu/duke/command/CalculateCapCommand.java

```diff
@@ -23,17 +23,25 @@ public void execute(SemesterList semesterList, AvailableModulesList availableMod
         Ui.showCap(cap);
     }
 
+    /** Calculate User's current Cumulative Average Point (CAP).
+     * @param semesterList All modules selected by user
+     */
     public void calculateCap(SemesterList semesterList) {
         double totalGradePoint = 0;
         int totalGradeModuleCredit = 0;
         for (SemModulesList sem : semesterList) {
             for (SelectedModule module : sem) {
-                if (!module.getDone()) {
+                boolean isModuleCompleted = module.getDone();
+                if (!isModuleCompleted) {
                     continue;
                 }
-                if (module.getGrade().getGrade().equals("F")) {
+                boolean isGradedModuleFailed = module.getGrade().getGrade().equals("F");
+                boolean isUngradedPassed = module.getGrade().getGrade().equals("CS");
+                boolean isUngradedFailed = module.getGrade().getGrade().equals("CU");
+                boolean isCompletedModuleGraded = !isUngradedPassed || !isUngradedFailed;
```

Comment:
Should the condition here be this instead?
```suggestion
                boolean isCompletedModuleGraded = !isUngradedPassed && !isUngradedFailed;
```
Since we want boolean to be true if its `!isUngradedPassed` and `!isungradedFailed`